platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java

```diff
@@ -660,24 +665,16 @@ boolean onTap(PointF tapPoint, float screenDensity) {
           if (annotation.getId() == newSelectedMarkerId) {
             Marker marker = (Marker) annotation;
 
-            if (marker instanceof MarkerView) {
-              handledDefaultClick = markerViewManager.onClickMarkerView((MarkerView) marker);
-            } else {
+            if (!(marker instanceof MarkerView)) {
```

Comment:
We should refactor `AnnotationManager` class (actually I'm working on it 😅 ). It has a lot of responsibilities.